src/handlers/auth/fs.ts

```diff
@@ -11,11 +11,18 @@ const fileName =
 
 const filePath = path.join(app.getPath('userData'), fileName)
 
+let profileDataCache: Profile | null = null
```

Comment:
You need to invalidate the cace in the `saveProfileData` function as well, or the values will go out of sync:

```ts
try {
  await writeFile(filePath, JSON.stringify(profile, null, 2))
} finally {
  // Force profile to be read from disk again
  profileDataCache = null
}
```